llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -1240,6 +1269,116 @@ struct FunctionStackPoisoner : public InstVisitor<FunctionStackPoisoner> {
 
 } // end anonymous namespace
 
+// Append a new argument "launch_data" to user's spir_kernels
```

Comment:
How  will it work at runtime? SYCL RT sets kernel arguments based on integration header information provided by SYCL FE compiler. There won't be any information about this new extra kernel argument and I don't think that it will be set.

Reply:
[`clEnqueueNDRangeKernel`](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clEnqueueNDRangeKernel) explicitly documents that there should be an error if kernel argument is not set:

> [`CL_INVALID_KERNEL_ARGS`](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_INVALID_KERNEL_ARGS) if the kernel argument values have not been specified.

I don't fully understand why and how it works then. Perhaps there is a bug in our own OpenCL CPU RT implementation and we don't emit that error.

I would recommend to at least add a `TODO` comment here to investigate, because it is a potential bug

Reply:
> [`clEnqueueNDRangeKernel`](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clEnqueueNDRangeKernel) explicitly documents that there should be an error if kernel argument is not set:
> 
> > [`CL_INVALID_KERNEL_ARGS`](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#CL_INVALID_KERNEL_ARGS) if the kernel argument values have not been specified.
> 
> I don't fully understand why and how it works then. Perhaps there is a bug in our own OpenCL CPU RT implementation and we don't emit that error.
> 
> I would recommend to at least add a `TODO` comment here to investigate, because it is a potential bug

Apologies for not explaining this clearly, we do set this new argument in UR.

Reply:
I mean SYCL doesn't know this new argument at all, but we know this and set it in our sanitizer layer of UR.